### PR TITLE
Feat/logs readable

### PR DESCRIPTION
**Pull request by Rrock-k**
## Logs
make logs human readable

Screenshot 2023-07-21 at 00 53 16

## Who won

Screenshot 2023-07-21 at 00 53 19

### CODE REVIEW

File: frontend/src/pages/Battle/Battle.tsx [hunk 1]
```diff
@@ -11,11 +11,56 @@ import { Button } from "components/Button";
...
+type Turn =
+  | TurnGeneric<"move", { position: number }>
+  | TurnGeneric<"attack", { damage: number; position: number }>
+  | TurnGeneric<"miss", { position: number }>
+  | TurnGeneric<"rest", { position: number }>;
+
+type BattleLogs = {
+  logs: Array<
+    {
+      [key in `c${"1" | "2" | "3" | "4"}`]: string;
```
Comment (tmcgroul) [inline frontend/src/pages/Battle/Battle.tsx:30]: `BattleLog` represents a battle between two gladiators so only c1 and c2 are possible values

**Review by tmcgroul — CHANGES_REQUESTED** [review]

File: frontend/src/pages/Battle/Battle.tsx (hunk 1 above)
Comment (Rrock-k) [reply frontend/src/pages/Battle/Battle.tsx]: fixed

File: frontend/src/pages/Battle/Battle.tsx [hunk 2]
```diff
@@ -11,11 +11,52 @@ import { Button } from "components/Button";
...
+  logs: Array<{
+    c1: string;
+    c2: string;
+    turns: Turn[];
+    winner: string;
+  }>;
+  winner: string;
+};
+
+const actions = {
```
Comment (wmrsmile2018) [inline frontend/src/pages/Battle/Battle.tsx:35]: think u should create a new file textMap.ts for this

File: frontend/src/pages/Battle/Battle.tsx [hunk 3]
```diff
@@ -11,11 +11,52 @@ import { Button } from "components/Button";
...
+
+type TurnTypes = "move" | "attack" | "miss" | "rest";
+type TurnGeneric<K extends TurnTypes, T> = { [key in K]: T };
+type Turn =
+  | TurnGeneric<"move", { position: number }>
+  | TurnGeneric<"attack", { damage: number; position: number }>
+  | TurnGeneric<"miss", { position: number }>
+  | TurnGeneric<"rest", { position: number }>;
+
+type BattleLogs = {
```
Comment (wmrsmile2018) [inline frontend/src/pages/Battle/Battle.tsx:25]: now we are using localstorage for battle info logs, so we need re export  this for another file
for example page My Logs (Profile)

Comment (Rrock-k) [reply frontend/src/pages/Battle/Battle.tsx]: done

File: frontend/src/pages/Battle/Battle.tsx (hunk 2 above)
Comment (Rrock-k) [reply frontend/src/pages/Battle/Battle.tsx]: done

**Pull request merged**